x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/engine/engine_search_preview/engine_search_preview.tsx

```diff
@@ -0,0 +1,230 @@
+import { EnterpriseSearchEnginesPageTemplate } from '../../layout/engines_page_template';
+
+import { EngineIndicesLogic } from '../engine_indices_logic';
+import { EngineViewLogic } from '../engine_view_logic';
+
+import { EngineSearchPreviewLogic } from './engine_search_preview_logic';
+
+class InternalEngineTransporter implements Transporter {
+  constructor(private http: HttpSetup, private engineName: string) {}
+
+  async performRequest(request: SearchReqest) {
+    const url = `/internal/enterprise_search/engines/${this.engineName}/search`;
+
+    const response = await this.http.post<SearchResponse>(url, {
+      body: JSON.stringify(request),
+    });
+
+    const withUniqueIds = {
+      ...response,
+      hits: {
+        ...response.hits,
+        hits: response.hits.hits.map((hit) => ({
+          ...hit,
+          _id: btoa(JSON.stringify([hit._index, hit._id])),
+        })),
+      },
+    };
+
+    return withUniqueIds;
+  }
+}
+
+const ResultsView: React.FC<ResultsViewProps> = ({ children }) => {
+  return <EuiFlexGroup direction="column">{children}</EuiFlexGroup>;
+};
+
+const ResultView: React.FC<ResultViewProps> = ({ result }) => {
+  const fields = Object.entries(result)
+    .filter(([key]) => !key.startsWith('_') && key !== 'id')
+    .map(([key, value]) => {
+      return {
+        name: key,
+        value: value.raw,
+      };
+    });
+
+  const {
+    _meta: {
+      id: encodedId,
+      rawHit: { _index: index },
+    },
+  } = result;
+
+  const [, id] = JSON.parse(atob(encodedId));
+
+  const columns: Array<EuiBasicTableColumn<SearchResult>> = [
+    {
+      field: 'name',
+      name: 'name',
+      render: (name: string) => {
+        return <code>&quot;{name}&quot;</code>;
+      },
+      truncateText: true,
+      width: '20%',
+    },
+    {
+      field: 'value',
+      name: 'value',
+      render: (value: string) => value,
+    },
+  ];
+
+  return (
+    <EuiPanel paddingSize="m">
+      <EuiFlexGroup direction="column">
+        <EuiFlexGroup justifyContent="spaceBetween">
+          <code>ID: {id}</code>
+          <EuiFlexItem grow={false}>
+            <EuiFlexGroup gutterSize="xs" alignItems="center">
+              <code>from</code>
+              <EuiBadge color="hollow">{index}</EuiBadge>
+            </EuiFlexGroup>
+          </EuiFlexItem>
+        </EuiFlexGroup>
+        <EuiBasicTable items={fields} columns={columns} />
+      </EuiFlexGroup>
+    </EuiPanel>
+  );
+};
+
+const InputView: React.FC<InputViewProps> = ({ getInputProps }) => {
+  return (
+    <EuiFlexGroup gutterSize="s">
+      <EuiFieldSearch
+        fullWidth
+        placeholder="search"
+        {...getInputProps({})}
+        isClearable
+        aria-label="Search Input"
+      />
+      <EuiButton type="submit" color="primary" fill>
+        Search
+      </EuiButton>
+    </EuiFlexGroup>
+  );
+};
+
+const SortingView: React.FC<SortingViewProps> = ({ options = [], value, onChange }) => {
+  return (
```

Comment:
as we haven't got any fields configured for sorting, we should omit this for now and create a separate issue targeting sorting fields.